packages/block-editor/src/components/block-styles/index.js

```diff
@@ -110,18 +110,21 @@ function BlockStyles( {
 		return null;
 	}
 
-	if ( ! type.styles && ! find( styles, 'isDefault' ) ) {
```

Comment:
Looks like you already knew it two years ago :) https://github.com/WordPress/gutenberg/pull/12519/files#r240045926

Reply:
Ah, so what happened was that by [setting](https://github.com/WordPress/gutenberg/pull/22849/files#diff-33e0ac9c44b16a617039a563d4c065f3R170) `[]` as default for `styles`, the `!type.styles` was no longer true. :+1: